runtime/Java/src/org/antlr/v4/runtime/ANTLRErrorListener.java

```diff
@@ -76,7 +78,8 @@
 	public void syntaxError(Recognizer<?, ?> recognizer,
 							Object offendingSymbol,
 							int line,
-							int charPositionInLine,
+							int startPositionInLine,
+							int endPositionInLine,
```

Comment:
:exclamation: This change assumes the token (or error) ends on the same line where recognition started. This is not always the case.